src/service/genesis_gen.h

```diff
@@ -420,6 +420,18 @@ namespace ccf
         {.status = CodeStatus::ALLOWED_TO_JOIN, .platform = platform});
     }
 
+    void trust_node_security_policy(
+      const std::optional<RawPolicy>& raw_security_policy, const DigestedPolicy& host_data)
+    {
+      auto security_policies = tx.rw(tables.security_policies);
+      if (raw_security_policy.has_value()){
+        LOG_INFO_FMT("Trusting node with policy {}", raw_security_policy.value());
+        security_policies->put(host_data, raw_security_policy.value());}
+      else{
+        LOG_INFO_FMT("Trusting node with unset policy");
+        security_policies->put(host_data, "");}
```

Comment:
It would be nice to have a new C++ type for the empty security policy case, rather than `""`.

Reply:
Would it still serialise to the empty string?

Reply:
Or it could be `static constexpr default_empty_security_policy = "";`.

Reply:
I've had a slight change of heart here, I had done this as of a68507926995c6e0ef747055e7dd0e8b1387c56b but that was an `snp::` field which then wouldn't compile for SGX, I also only really used it here so I don't think it's strictly necessary?

Reply:
Fixing SGX builds allowed me to bring this back